source/github-helpers/search-query.ts

```diff
@@ -35,102 +33,102 @@ export default class SearchQuery {
 		return value.includes(' ') ? `"${value}"` : value;
 	}
 
-	link?: HTMLAnchorElement;
-	searchParams: URLSearchParams;
-
-	constructor(link: Source) {
-		if (link instanceof HTMLAnchorElement) {
-			this.link = link;
-			this.searchParams = new URLSearchParams(link.search);
-			// Keep `.search` property up to date with this `searchParams`
-			const nativeSet = this.searchParams.set;
-			this.searchParams.set = (name, value) => {
-				nativeSet.call(this.searchParams, name, value);
-				link.search = String(this.searchParams);
-			};
-		} else if (link instanceof URL) {
-			this.searchParams = link.searchParams;
-		} else {
-			this.searchParams = new URLSearchParams(link);
+	static from(link: HTMLAnchorElement): SearchQuery {
```

Comment:
This can accept `Location` too, and maybe more.
```suggestion
	static from(link: Source): SearchQuery {
```

Perhaps you can merge `fromSearchParams` back into it